### PR TITLE
 Remove Wikipedia Zero, remove AFNetworking 

### PR DESCRIPTION
**Pull request by joewalsh**
https://phabricator.wikimedia.org/T197528

### CODE REVIEW

File: Wikipedia/Code/EditPreviewViewController.m [hunk 1]
```diff
@@ -0,0 +1,100 @@
+#import "EditPreviewViewController.h"
```
Comment (nambatee) [inline Wikipedia/Code/EditPreviewViewController.m]: are this file + its header supposed to be here? looks like there's a `EditPreviewViewController.swift` below

File: Wikipedia/Code/EditSaveViewController.m [hunk 2]
```diff
@@ -0,0 +1,579 @@
+#import "EditSaveViewController.h"
```
Comment (nambatee) [inline Wikipedia/Code/EditSaveViewController.m]: same question as above - I don't see either of these old edit & preview files in Xcode but for some reason GitHub thinks they're coming in

Comment (joewalsh) [reply Wikipedia/Code/EditSaveViewController.m]: fixed - these were converted to swift in the other PR

File: Wikipedia/Code/SectionEditorViewController.swift [hunk 3]
```diff
@@ -145,10 +146,40 @@ class SectionEditorViewController: UIViewController {
         }
         let message = WMFLocalizedString("wikitext-downloading", value: "Loading content...", comment: "Alert text shown when obtaining latest revision of the section being edited")
         WMFAlertManager.sharedInstance.showAlert(message, sticky: true, dismissPreviousAlerts: true)
-        let manager = QueuesSingleton.sharedInstance()?.sectionWikiTextDownloadManager
-        QueuesSingleton.sharedInstance()?.sectionWikiTextDownloadManager.wmf_cancelAllTasks {
-            let _ = WikiTextSectionFetcher(andFetchWikiTextFor: section, with: manager, thenNotify: self)
+        sectionFetcher.fetch(section) { (result, error) in
+            if let error = error {
+                WMFAlertManager.sharedInstance.showErrorAlert(error as NSError, sticky: true, dismissPreviousAlerts: true)
```
Comment (nambatee) [inline Wikipedia/Code/SectionEditorViewController.swift]: should this and the other `showAlert` be dispatched to the main thread?

Comment (joewalsh) [reply Wikipedia/Code/SectionEditorViewController.swift]: yup, fixed

File: Wikipedia/Code/Session.swift [hunk 4]
```diff
@@ -261,24 +267,24 @@ import Foundation
         return task
     }
 
-    public func jsonDecodableTask<T: Decodable>(with url: URL?, method: Session.Request.Method = .get, bodyParameters: Any? = nil, bodyEncoding: Session.Request.Encoding = .json, authorized: Bool? = nil, completionHandler: @escaping (_ result: T?, _ response: URLResponse?,_ authorized: Bool?,  _ error: Error?) -> Swift.Void) {
```
Comment (nambatee) [inline Wikipedia/Code/Session.swift]: Apple's example in the Type Constraint Syntax part of The Swift Programming Language (4.2) https://docs.swift.org/swift-book/LanguageGuide/Generics.html shows the `<T: Decodable>` syntax as an example, I think it's more readable to have it up front, especially if the signature is long

Comment (joewalsh) [reply Wikipedia/Code/Session.swift]: good point - fixed

File: Wikipedia/Code/WMFArticleFetcher.m [hunk 5]
```diff
@@ -80,35 +63,25 @@ - (instancetype)initWithDataStore:(MWKDataStore *)dataStore {
 }
 
 - (void)dealloc {
-    [self.operationManager invalidateSessionCancelingTasks:YES];
-    [self.pageSummarySessionManager invalidateSessionCancelingTasks:YES];
+    [self cancelAllFetches];
```
Comment (nambatee) [inline Wikipedia/Code/WMFArticleFetcher.m]: `WMFLegacyFetcher` calls `cancelAllFetches` in its `dealloc` so I guess this could be deleted?

Comment (joewalsh) [reply Wikipedia/Code/WMFArticleFetcher.m]: yup! done

**Pull request merged**